cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/GlobalParam.scala

```diff
@@ -0,0 +1,161 @@
+package com.microsoft.azure.synapse.ml.services.openai
+
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import com.microsoft.azure.synapse.ml.services.HasServiceParams
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.Row
+
+import scala.collection.mutable
+
+sealed trait GlobalParamKey[T]
+case object OpenAIDeploymentNameKey extends GlobalParamKey[String]
+
+object GlobalParamObject {
```

Comment:
nit: perhaps place in core unless ServiceParam messes with your ability to do that

Reply:
I can't move it to core because of HasServiceParams, but moved it to `package com.microsoft.azure.synapse.ml.services`